src/java.base/share/classes/java/lang/reflect/code/op/ExtendedOps.java

```diff
@@ -3145,46 +3142,66 @@ public static PatternOps.RecordPatternOp recordPattern(RecordTypeDesc recordDesc
     }
 
     @OpDeclaration(StringTemplateOp.NAME)
-    public static final class StringTemplateOp extends OpWithDefinition {
+    public static final class StringTemplateOp extends OpWithDefinition implements Op.Nested {
 
         public static final String NAME = "java.stringTemplate";
 
+        private final Value processorValue;
+        private final List<Value> literalsValues;
+        private final List<Body> expressionsBodies;
+
         public StringTemplateOp(OpDefinition def) {
             super(def);
+
+            this.processorValue = operands().get(0);
+            this.literalsValues = operands().subList(1, operands().size());
+            this.expressionsBodies = def.bodyDefinitions().stream().map(bd -> bd.build(this)).toList();
         }
 
         StringTemplateOp(StringTemplateOp that, CopyContext cc, OpTransformer ot) {
             super(that, cc);
+
+            this.processorValue = operands().get(0);
+            this.literalsValues = operands().subList(1, operands().size());
+            this.expressionsBodies = that.expressionsBodies.stream()
+                    .map(b -> b.transform(cc, ot).build(this)).toList();
+        }
+
+        public StringTemplateOp(Value processorValue, List<Value> literalsValues, List<Body.Builder> expressionsBodies) {
+            super(NAME, makeOperandsList(processorValue, literalsValues));
```

Comment:
Add a comment before that to say we can use statements before super (JEP 447) when the compiler can depend on 22 features.